### PR TITLE
gRPC Web

### PR DESCRIPTION
**Pull request by sanderpick**
- Integrates https://github.com/improbable-eng/grpc-web which wraps the gRPC server.
- Adds the associated TS plugin to `pb/Makefile`.

### CODE REVIEW

File: pb/Makefile [hunk 1]
```diff
@@ -1,14 +1,19 @@
 PB = $(wildcard *.proto)
 GO = $(PB:.proto=.pb.go)
+PROTOC_GEN_TS_PATH = "./node_modules/.bin/protoc-gen-ts"
 
 all: $(GO)
 
 %.pb.go: %.proto
-	protoc -I=. -I=$(GOPATH)/src --gogofaster_out=\
+	protoc -I=. -I=$(GOPATH)/src \
```
Comment (carsonfarmer) [inline pb/Makefile]: yesssss!

File: pb/threads_pb.d.ts [hunk 2]
```diff
@@ -0,0 +1,480 @@
+// package: threads.pb
```
Comment (carsonfarmer) [inline pb/threads_pb.d.ts]: Are we sure we want to check these into version control? Maybe good for now, but might be better to just export and publish as a package at some point right?

File: threads.go [hunk 3]
```diff
@@ -100,13 +110,54 @@ func NewThreads(
 	go t.rpc.Serve(listener)
 	pb.RegisterThreadsServer(t.rpc, t.service)
 
+	webrpc := grpcweb.WrapServer(t.rpc)
```
Comment (carsonfarmer) [inline threads.go]: Awesome, glad this integrated so nicely.

File: pb/threads_pb.d.ts (hunk 2 above)
Comment (sanderpick) [reply pb/threads_pb.d.ts]: Totally. Let's see how these custom gogo types play with JS and go from there... I can then pull over the npm make-and-publish-on-tag stuff from `go-textile`

**Pull request merged**